pkg/event/target/mqtt.go

```diff
@@ -140,8 +156,25 @@ func NewMQTTTarget(id string, args MQTTArgs) (*MQTTTarget, error) {
 
 	client := mqtt.NewClient(options)
 	token := client.Connect()
-	if token.Wait() && token.Error() != nil {
-		return nil, token.Error()
+
+	if args.QueueDir != "" {
+		go func() {
+			// Repeat the pings until the client registers the token.
+			for {
+				if token.Wait() && token.Error() == nil {
+					fmt.Println("MQTT: Connected")
+					break
+				}
+				fmt.Println("MQTT: " + token.Error().Error() + ". Retrying in " + strconv.Itoa(retryTimeout) + " seconds.")
+				time.Sleep(retryTimeout * time.Second)
+				token = client.Connect()
```

Comment:
How about connection retry with store directory transparent inside `paho.mqtt.golang`?

Reply:
Yes the library did not have anything apis on retrying the connection. Their `SetReconnectInterval` did not work in our case. Shall we add an API (say, `RetryConnect`) and move this inside that ?

Reply:
I thought this is more client specific and did not have a plan to move it inside the library

Reply:
Done :+1: